Mathlib/NumberTheory/FLT/MasonStothers.lean

```diff
@@ -0,0 +1,253 @@
+/-
+Copyright (c) 2024 Jineon Back and Seewoo Lee. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jineon Baek, Seewoo Lee
+-/
+import Init.Data.Nat.Lemmas
+import Mathlib.Algebra.EuclideanDomain.Defs
+import Mathlib.Algebra.Polynomial.RingDivision
+import Mathlib.Algebra.Ring.Regular
+import Mathlib.RingTheory.Polynomial.Wronskian
+import Mathlib.RingTheory.Radical
+import Mathlib.RingTheory.UniqueFactorizationDomain
+import Mathlib.Algebra.Polynomial.FieldDivision
+
+/-!
+# Mason-Stothers theorem
+
+This file states and proves Mason-Stothers theorem, which is a polynomial version of ABC conjecture.
+For a (pairwise) coprime polynomials `a, b, c` (over a field) with `a + b + c = 0`, we have
+`max {deg(a), deg(b), deg(c)} + 1 ≤ deg(rad(abc))` or `a' = b' = c' = 0`.
+
+Proof is based on this online note by Franz Lemmermeyer http://www.fen.bilkent.edu.tr/~franz/ag05/ag-02.pdf,
+which is essentially based on Noah Snyder's paper "An Alternative Proof of Mason's Theorem",
+but slightly different.
+
+## TODO
+Prove polynomial FLT using Mason-Stothers theorem.
+-/
+
+noncomputable section
+
+open scoped Polynomial Classical
+
+open Polynomial UniqueFactorizationMonoid
+
+variable {k : Type*} [Field k]
+
+
+/-- Auxiliary theorems for `max₃ a b c = max (max a b) c`. -/
+def max₃ (a b c : Nat) : Nat :=
```

Comment:
Do we want this auxiliary function for maximum of three natural numbers in the statement of Mason--Stothers? I wonder how other parts of mathlib represents maximum values of three numbers.

Reply:
We usually just write `max (max a b) c`, or `a ⊔ b ⊔ c` if you prefer infixes. But also you are using it to state `max (max a b) c ≤ d`, in which situation we much prefer writing `a ≤ d ∧ b ≤ d ∧ c ≤ d`

Reply:
Should we remove all `max₃` related apis and work with `max`-related things then?

Reply:
Yes that would great 🙏